tests/unit/browser/test_history.py

```diff
@@ -205,60 +205,76 @@ def raise_error(url, replace=False):
             with pytest.raises(sql.BugError):
                 web_history.add_url(QUrl('https://www.example.org/'))
 
-    @pytest.mark.parametrize('level, url, req_url, expected', [
-        (logging.DEBUG, 'a.com', 'a.com', [('a.com', 'title', 12345, False)]),
-        (logging.DEBUG, 'a.com', 'b.com', [('a.com', 'title', 12345, False),
-                                           ('b.com', 'title', 12345, True)]),
-        (logging.WARNING, 'a.com', '', [('a.com', 'title', 12345, False)]),
-        (logging.WARNING, '', '', []),
-        (logging.WARNING, 'data:foo', '', []),
-        (logging.WARNING, 'a.com', 'data:foo', []),
+    @pytest.mark.parametrize('level, url, expected', [
+        (logging.DEBUG, 'a.com', [('a.com', 'title', 12345, False)]),
+        (logging.WARNING, '', []),
+        (logging.WARNING, 'data:foo', []),
+        (logging.WARNING, 'qute://back', []),
     ])
     def test_from_tab(self, web_history, caplog, mock_time,
-                      level, url, req_url, expected):
+                      level, url, expected):
         with caplog.at_level(level):
-            web_history.add_from_tab(QUrl(url), QUrl(req_url), 'title')
+            web_history.add_from_tab(QUrl(url), 'title',)
         assert set(web_history) == set(expected)
 
     def test_exclude(self, web_history, config_stub):
         """Excluded URLs should be in the history but not completion."""
         config_stub.val.completion.web_history.exclude = ['*.example.org']
         url = QUrl('http://www.example.org/')
-        web_history.add_from_tab(url, url, 'title')
+        web_history.add_from_tab(url, 'title')
         assert list(web_history)
         assert not list(web_history.completion)
 
-    def test_no_immedate_duplicates(self, web_history, mock_time):
```

Comment:
Why remove this test, you are preserving this behaviour currently right?

Reply:
I'm changing the behaviour slightly to operate on a per-tab basis. The existing test assumes that deduplication happens globally within the WebHistory class.